src/Evolution/DgSubcell/Tags/Jacobians.hpp

```diff
@@ -33,6 +35,122 @@ struct InverseJacobianLogicalToGrid : db::SimpleTag {
 /// and reduce the memory footprint.
 struct DetInverseJacobianLogicalToGrid : db::SimpleTag {
   static std::string name() { return "Det(InverseJacobian(Logical,Grid))"; }
-  using type = std::optional<Scalar<DataVector>>;
+  using type = Scalar<DataVector>;
 };
+
+/// \brief The inverse Jacobian from the element logical frame to the inertial
+/// frame at the cell centers.
+///
+/// Specifically, \f$\partial x^{\bar{i}} / \partial x^i\f$, where \f$\bar{i}\f$
+/// denotes the element logical frame and \f$i\f$ denotes the inertial frame.
+///
+/// \note stored as a `std::optional` so we can reset it when switching to DG
+/// and reduce the memory footprint (is it useful for a ComputeTag though?).
+template <size_t Dim>
+struct InverseJacobianLogicalToInertial : db::SimpleTag {
+  static std::string name() { return "InverseJacobian(Logical,Inertial)"; }
+  using type = ::InverseJacobian < DataVector, Dim, Frame::ElementLogical,
+        Frame::Inertial >;
+};
+
+/// Compute item for the inverse jacobian matrix from logical to
+/// grid coordinates
+template <typename MapTagLogicalToGrid, size_t Dim>
+struct InverseJacobianLogicalToGridCompute : InverseJacobianLogicalToGrid<Dim>,
+                                             db::ComputeTag {
+  static constexpr size_t dim = Dim;
+  using base = InverseJacobianLogicalToGrid<Dim>;
+  using return_type = typename base::type;
+  using argument_tags = tmpl::list<
+      MapTagLogicalToGrid,
+      evolution::dg::subcell::Tags::Coordinates<dim, Frame::ElementLogical>>;
+  static void function(
+      const gsl::not_null<return_type*> inverse_jacobian_logical_to_grid,
+      const ElementMap<Dim, Frame::Grid>& logical_to_grid_map,
+      const tnsr::I<DataVector, dim, Frame::ElementLogical>& logical_coords) {
+    *inverse_jacobian_logical_to_grid =
+        logical_to_grid_map.inv_jacobian(logical_coords);
+  }
+};
+
+/// Compute item for the determinant of the inverse jacobian matrix
+/// from logical to grid coordinates
+template <size_t Dim>
+struct DetInverseJacobianLogicalToGridCompute
+    : DetInverseJacobianLogicalToGrid,
+      db::ComputeTag {
+  static constexpr size_t dim = Dim;
+  using base = DetInverseJacobianLogicalToGrid;
+  using return_type = typename base::type;
+  using argument_tags = tmpl::list<InverseJacobianLogicalToGrid<Dim>>;
+  static void function(
+      const gsl::not_null<return_type*> det_inverse_jacobian_logical_to_grid,
+      const ::InverseJacobian<DataVector, Dim, Frame::ElementLogical,
+                              Frame::Grid>
+          inverse_jacobian_logical_to_grid) {
+    *det_inverse_jacobian_logical_to_grid =
+        determinant(inverse_jacobian_logical_to_grid);
+  }
+};
+
+/// Compute item for the inverse jacobian matrix from logical to
+/// inertial coordinates
+template <typename MapTagGridToInertial, size_t Dim>
```

Comment:
Can you also add the test you have in the file `tests/Unit/Evolution/Systems/GrMhd/GhValenciaDivClean/Subcell/Test_JacobianCompute.cpp` to the PR? I think we need to test our compute tags more for functionality. I started doing this for subcell and it helped find a lot of bugs...